x/ccv/provider/types/codec.go

```diff
@@ -46,6 +46,14 @@ func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
 		(*sdk.Msg)(nil),
 		&MsgSubmitConsumerDoubleVoting{},
 	)
+	registry.RegisterImplementations(
+		(*sdk.Msg)(nil),
+		&MsgOptIn{},
+	)
+	registry.RegisterImplementations(
+		(*sdk.Msg)(nil),
+		&MsgOptOut{},
```

Comment:
Shouldn't we register also MsgSetConsumerCommissionRate?

Reply:
Good catch.